src/IO/InputStreamFromROOTFile.cxx

```diff
@@ -127,42 +128,88 @@ InputStreamFromROOTFile::set_up(const std::string & header_path)
-        stream_ptr->SetBranchAddress("sourcePosZ2",&sourcePosZ2);
+        stream_ptr->SetBranchAddress("axialPos",&axialPos, &br_axialPos);
+        stream_ptr->SetBranchAddress("globalPosX1",&globalPosX1, &br_globalPosX1);
+        stream_ptr->SetBranchAddress("globalPosX2",&globalPosX2, &br_globalPosX2);
+        stream_ptr->SetBranchAddress("globalPosY1",&globalPosY1, &br_globalPosY1);
+        stream_ptr->SetBranchAddress("globalPosY2",&globalPosY2, &br_globalPosY2);
+        stream_ptr->SetBranchAddress("globalPosZ1",&globalPosZ1, &br_globalPosZ1);
+        stream_ptr->SetBranchAddress("globalPosZ2",&globalPosZ2, &br_globalPosZ2);
+        stream_ptr->SetBranchAddress("rotationAngle",&rotation_angle, &br_rotation_angle);
+        stream_ptr->SetBranchAddress("runID",&runID, &br_runID);
+        stream_ptr->SetBranchAddress("sinogramS",&sinogramS, &br_sinogramS);
+        stream_ptr->SetBranchAddress("sinogramTheta",&sinogramTheta, &br_sinogramTheta);
+        stream_ptr->SetBranchAddress("sourceID1",&sourceID1, &br_sourceID1);
+        stream_ptr->SetBranchAddress("sourceID2",&sourceID2, &br_sourceID2);
+        stream_ptr->SetBranchAddress("sourcePosX1",&sourcePosX1, &br_sourcePosX1);
+        stream_ptr->SetBranchAddress("sourcePosX2",&sourcePosX2, &br_sourcePosX2);
+        stream_ptr->SetBranchAddress("sourcePosY1",&sourcePosY1, &br_sourcePosY1);
+        stream_ptr->SetBranchAddress("sourcePosY2",&sourcePosY2, &br_sourcePosY2);
+        stream_ptr->SetBranchAddress("sourcePosZ1",&sourcePosZ1, &br_sourcePosZ1);
+        stream_ptr->SetBranchAddress("sourcePosZ2",&sourcePosZ2, &br_sourcePosZ2);
     }
 
     return Succeeded::yes;
 }
 
+bool
+InputStreamFromROOTFile::check_brentry_randoms_scatter_energy_conditions(Long64_t brentry)
+{
+  if (brentry < 0)
+    return false;
+
+  // Scatter event condition.
+  if (this->exclude_scattered) {
+    if (br_comptonPhantom1->GetEntry(brentry) == 0)
+      return false;
+    if (br_comptonPhantom2->GetEntry(brentry) == 0)
+      return false;
+    // Check if either event has been Compton scattered
+    if (this->comptonphantom1 > 0 || this->comptonphantom2 > 0)
+      return false;
+  }
+
+  // Random event condition.
+  if (this->exclude_randoms) {
+    if (br_eventID1->GetEntry(brentry) == 0)
+      return false;
+    if (br_eventID2->GetEntry(brentry) == 0)
+      return false;
+    // Check for the same event
+    if ((this->eventID1 != this->eventID2))
+      return false;
+  }
+
+  // Energy condition.
+  if (br_energy1->GetEntry(brentry) == 0)
+    return false;
+  if (br_energy2->GetEntry(brentry) == 0)
+    return false;
+  // Check both energy values are within window
+  if (this->get_energy1_in_keV() < this->low_energy_window ||
+      this->get_energy1_in_keV() > this->up_energy_window ||
+      this->get_energy2_in_keV() < this->low_energy_window ||
+      this->get_energy2_in_keV() > this->up_energy_window)
+  {
+    return false;
+  }
```

Comment:
Can we have an if around if energy information is default we can ignore these tests.

Reply:
What does a "default" mean? It might not be the same for GATE macro as STIR. I guess you could introduce yet another `bool check_energy_window_information` defaulting to `true`, but I feel it's creating work for not much benefit. Maybe I'm wrong.

Reply:
Yes, we can (and probably should) do this. By default I meant the class defaults:
https://github.com/UCL/STIR/blob/a08cb7504aac957742e2b7a81240a38a62f2e489/src/IO/InputStreamFromROOTFile.cxx#L71-L72

Reply:
>but I feel it's creating work for not much benefit. Maybe I'm wrong.

It was easy.